airflow/hooks/webhdfs_hook.py

```diff
@@ -0,0 +1,81 @@
+from airflow.hooks.base_hook import BaseHook
+from airflow.configuration import conf
+import logging
+
+from hdfs import InsecureClient, HdfsError
+from airflow.utils import AirflowException
+
+
+class WebHDFSHookException(AirflowException):
+    pass
+
+
+class WebHDFSHook(BaseHook):
+    """
+    Interact with HDFS. This class is a wrapper around the hdfscli library.
+    """
+    def __init__(self, webhdfs_conn_id='webhdfs_default'):
+        self.webhdfs_conn_id = webhdfs_conn_id
+
+    def get_conn(self):
+        """
+        Returns a hdfscli InsecureClient object.
+        """
+        nn_connections = self.get_connections(self.webhdfs_conn_id)
+        for nn in nn_connections:
+            try:
```

Comment:
Not necessarily. It is a poor man's High Availability client so that we can enter a list of the namenodes (primary and secondary) in the connections. It only raises if all the namenodes with that connection ID fail. Are you suggesting we cache it and then rescue the specific error and retry with another namenode in the list if needed?